### PR TITLE
new(tests): add state test with empty 7702 authorization list

### PR DESCRIPTION
**Pull request by chfast**
## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [ ] All: Considered squashing commits to improve commit history.
- [ ] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [ ] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: tests/prague/eip7702_set_code_tx/test_set_code_txs.py [hunk 1]
```diff
@@ -2943,6 +2943,28 @@ def test_contract_create(
...
+def test_empty_authorization_list(
+    state_test: StateTestFiller,
+    pre: Alloc,
+):
+    """Test sending an invalid transaction with empty authorization list."""
+    tx = Transaction(
+        gas_limit=100_000,
+        to=pre.deploy_contract(code=b""),
+        value=0,
+        authorization_list=[],
```
Comment (marioevz) [inline tests/prague/eip7702_set_code_tx/test_set_code_txs.py:2955]: I vaguely remember @lightclient mentioning that this test would fail on the tx decoding part, which leads to failure of the state test. Tagging him to help us out on whether including this test would pose an issue.

Comment (chfast) [reply tests/prague/eip7702_set_code_tx/test_set_code_txs.py:2955]: So this is ok then, or are you executing an invalid transaction?

Comment (winsvega) [reply tests/prague/eip7702_set_code_tx/test_set_code_txs.py:2955]: looks like a case for transaction tests

**Review by winsvega — APPROVED** [review]

File: tests/prague/eip7702_set_code_tx/test_set_code_txs.py (hunk 1 above)
Comment (marioevz) [reply tests/prague/eip7702_set_code_tx/test_set_code_txs.py:2942]: Just checked on `evm` command and the state test and blockchain test both correctly reject the transaction.

**Review by marioevz — APPROVED** [review]
Already rebased and added the EELS exception for test generation. Will merge once the CI passes.
